prime-router/src/main/kotlin/fhirengine/azure/FHIRFlowFunctions.kt

```diff
@@ -54,8 +61,20 @@ class FHIRFlowFunctions : Logging {
             responseBuilder.body(body)
         } catch (e: IllegalArgumentException) {
             responseBuilder.body(e.message)
+            actionHistory.trackLogs(
+                ActionLog(InvalidTranslationMessage(e.message ?: "Translation failure."), type = ActionLogLevel.error)
+            )
         }
-        return responseBuilder.build()
+
+        val workflowEngine = WorkflowEngine.Builder().metadata(Metadata.getInstance()).build()
```

Comment:
Don't use this.  Use the singleton or existing instance of Workflow engine.  That builder is for unit tests only.

Reply:
Actually, change this to be a constructor argument like in ReportFunction, so we can do dependency injection.

Reply:
I second Carlos' suggestion to use the constructor argument so we support DI.

Reply:
As per the PR review, I fixed it and it's now set like in Report.